### PR TITLE
[VxAdmin] Move CVR file mode logic to backend

### PR DESCRIPTION
**Pull request by kofi-q**
## Overview
Related to: https://github.com/votingworks/vxsuite/issues/2716

- Move the logic for locking to a "test" or "live" CVR file mode over to the backend.
- Currently maintains parity by searching for any previously imported CVRs and checking the `_testBallot` flag on one of them.
- Also moving the enum for CVR file mode to the backend and adding an "unlocked" option to represent the state where no CVR files have been imported yet. This was mostly to satisfy `react-query`'s requirement that query data can't resolve to `undefined`, but the explicitness seems helpful for readability anyway.
- Slight change to `CastVoteRecordFiles` to stop relying on just file name to determine if a CVR file has "live" or "test" records. Instead, reading the first line to determine mode.

Not sure I've unearthed all ripple effects, so let me know if I've missed anything.

Also let me know if I haven't matched any code/test patterns properly.

## Demo Video or Screenshot

https://user-images.githubusercontent.com/264902/199115279-9bb3beba-b4da-4674-b233-6a32cb72d05f.mov

## Testing Plan 
- Manually tested any relevant UI pieces I could find, based on wherever the old `fileMode` field was referenced.
- Added tests for new code, updated existing tests for modified code

## Checklist
- [n/a] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.
- [x] I have added JSDoc comments to any newly introduced exports

- [x] I have added a screenshot and/or video to this PR to demo the change
- [n/a] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: frontends/election-manager/src/components/export_batch_tally_results_button.tsx [hunk 1]
```diff
@@ -4,18 +4,20 @@ import {
...
   generateBatchResultsDefaultFilename,
 } from '@votingworks/utils';
+import { Admin } from '@votingworks/api';
 import { SaveFileToUsb, FileType } from './save_file_to_usb';
 import { AppContext } from '../contexts/app_context';
 import { generateBatchTallyResultsCsv } from '../utils/generate_batch_tally_results_csv';
+import { useCvrFileModeQuery } from '../hooks/use_cvr_file_mode_query';
 
-export function ExportBatchTallyResultsButton(): JSX.Element {
+export function ExportBatchTallyResultsButton(): JSX.Element | null {
```
Comment (eventualbuddha) [inline frontends/election-manager/src/components/export_batch_tally_results_button.tsx:13]: Unnecessary change.
```suggestion
export function ExportBatchTallyResultsButton(): JSX.Element {
```

**Review by eventualbuddha — APPROVED** [review]
A couple small things, but overall looks good.

File: frontends/election-manager/src/utils/cast_vote_record_files.ts [hunk 2]
```diff
@@ -200,19 +198,19 @@ export class CastVoteRecordFiles {
             importedCvrCount: importedFile.importedCvrCount,
             scannerIds: importedFile.scannerIds,
             exportTimestamp: importedFile.exportTimestamp,
-            isTestModeResults: this.fileMode === 'test',
+            isTestModeResults: importedFile.allCastVoteRecords[0]._testBallot,
```
[the remark this answers is not in the document]
Comment (kofi-q) [reply frontends/election-manager/src/utils/cast_vote_record_files.ts:201]: ah, good catch - will look at handling that/working the defaulting logic back in
